File: retail-managment-system/rms-ui/src/app/domain/order/models/order.ts
import { Customers } from "../../customers/models/customers";
import { Employee } from "../../employee/models/employee";
import { Installment } from "../../installment/models/installment";
import { orderItem } from "./orderItem";
import { paymentEnum } from "./paymentEnum";

export interface Order {
    id: number;
    version: number;
    orderDate: Date;
    paymentType: paymentEnum;
    paidAmount: number;
    remainingAmount: number;
    installmentAmount?: Installment;
    customer : Customers;
    totalPrice:number;
    orderItems:orderItem;
    employee:Employee;
}

File: retail-managment-system/rms-api/src/main/java/com/rms/domain/sales/Order.java
package com.rms.domain.sales;

import com.rms.domain.common.AuditingEntity;
import com.rms.domain.core.Employee;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "Orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Order extends AuditingEntity {

    @Temporal(TemporalType.DATE)
    @Column(name = "order_date")
    private Date orderDate;

    @Column(name = "payment_type")
    @Enumerated(EnumType.STRING)
    private PaymentType paymentType;

    @Column(name = "paid_amount")
    private Double paidAmount;

    @Column(name = "total_price")
    private Double totalPrice;

    @Column(name = "remaining_amount")
    private Double remainingAmount;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id")
    private Employee employee;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
    private Customer customer;

}
